apps/backend/src/auth/auth.service.ts

```diff
@@ -0,0 +1,119 @@
+import { AuthSchProfile } from '@kir-dev/passport-authsch';
+import { Injectable } from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import { PrismaService } from 'nestjs-prisma';
+import { User } from 'src/users/entities/user.entity';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private prisma: PrismaService,
+    private jwtService: JwtService
+  ) {}
+
+  login(user: User): string {
+    return this.jwtService.sign(user, {
+      secret: process.env.JWT_SECRET,
+      expiresIn: '7 days',
+    });
+  }
+
+  async findOrCreateUser(userProfile: AuthSchProfile): Promise<User> {
+    const user = await this.prisma.user.findUnique({ where: { authSchId: userProfile.authSchId } });
+    if (user) {
+      return user;
+    }
+
+    return this.prisma.user.create({
+      data: {
+        authSchId: userProfile.authSchId,
+        fullName: userProfile.fullName,
+        email: userProfile.email,
+        phone: userProfile.phone,
+      },
+    });
+  }
+
+  async syncClubMembership(user: User, userProfile: AuthSchProfile): Promise<User> {
```

Comment:
Couldn't there be a problem with this if PÉK contains depricated information? For example, I'm currently a gatekeeper in MMMK, but on PÉK I don't have the gatekeeper role.